### ISSUE TITLE
Feature request: never validate not-required blank/empty/unchecked inputs

### ISSUE THREAD

**Issue by indrimuska** [bvaughn/angular-form-for#148]
Every time an input is set to **not-required**, I would expect the plugin doesn't validate the field when it is blank (empty) even if it has a custom validator.

Let me make the problem clear.

Let's have a simple email input that accepts only gmail accounts ([plunker](http://plnkr.co/edit/w9kqIQa6U1cESUVBsJNr?p=preview)):
```javascript
email: {
   inputType: 'text',
   placeholder: 'please, provide a gmail account',
   required: false,
   pattern: /\w+@gmail\.com$/
}
```
If I leave the input blank and I try to submit the form, the plugin prevents submission and shows the classic **Invalid format** error.

![schermata 2015-07-28 alle 22 58 23](https://cloud.githubusercontent.com/assets/1561134/8943816/fdbd428c-357c-11e5-84c4-9d3f9eedeb28.png)

The question I made when I see that was: **why?** The input is not-required, so why the plugin validates it? As a normal HTML5 email input (with no `required` attribute) is not "processed" by the browser (or by Angular) when I leave it blank, the same I expect from this plugin.

**Comment by bvaughn** [comment]
You make a strong argument. I'll try to implement this change in the next week or so.

**Comment by bvaughn** [comment]
This will go out in the next release.

**Comment by bvaughn** [comment]
This is available in version [4.1.0](https://github.com/bvaughn/angular-form-for/releases/tag/4.1.0)

**Comment by indrimuska** [comment]
Great, this works like a charm! :+1:
